paywall/src/middlewares/web3Middleware.js

```diff
@@ -116,12 +120,31 @@ export default function web3Middleware({ getState, dispatch }) {
         }
       }
 
-      if (action.type === ADD_TRANSACTION) {
```

Comment:
I would keep  `ADD_TRANSACTION` and `NEW_TRANSACTION` distinct because the behavior is not strictly the same...